psi4/driver/driver_nbody.py

```diff
@@ -25,830 +25,1290 @@
 #
 # @END LICENSE
 #
-
+"""Plan, run, and assemble QC tasks to obtain many-body expansion and basis-set superposition error treatments.
+
+=============
+ManyBody Flow
+=============
+Bullet points are major actions
+Lines of dashes denote function calls
+e/d/dd=dg/g/h := energy, dipole, dipole derivative = dipole gradient, gradient, Hessian
+
+---------------------------
+ManyBodyComputer.__init__()
+---------------------------
+* not an explicit function but pydantic handles some defaults and validation
+* fields molecule, nfragments, bsse_type, return_total_data, and initial max_nbody set
+* BaseComputer.__init__()
+
+task_planner()
+--------------
+* computer gets modified from task_planner outside this file!
+* user levels kwarg processed into modelchem treatment levels and the n-body levels they cover. fields nbodies_per_mc_level set and max_nbody reset
+* for each modelchem treatment level, call build_tasks() below via one of four routes, depending on simple MB or layered MB(FD), MB(CBS), or MB(FD(CBS))
+
+    ------------------------------
+    ManyBodyComputer.build_tasks()
+    ------------------------------
+    * if supersystem requested, request (frag, bas) indices for full nbody range of nocp treatment from build_nbody_compute_list()
+    * otherwise, request (frag, bas) indices for specified nbody range covering specified bsse treatments from build_nbody_compute_list()
+
+        build_nbody_compute_list()
+        --------------------------
+        * initializes dicts for each of nocp, cp, vmfc (2 for this one) with keys requested n-body levels and values empty sets
+        * use combinatorics formulas to fill each key with (frag, bas) indices (what fragments are active and what fragments have basis functions)
+          needed to compute the requested bsse treatments at the requested n-body levels.
+        * merge by n-body level the sets of indices for each bsse treatment into an "all" dict. return this and all the per-bsse dicts.
+
+    * merge all the requested indices, prepend a modelchem treatment index to form `mc_(frag, bas)`
```

Comment:
What does "merge indices" mean?

Reply:
you have lists of needed (frag, bas) for all the requested bsse_types, so form set union